tests/integration/test_valid_data_in_request.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: MIT
+
+import utils
+import time
+
+
+def test_valid_data_in_request(test_env):
+    """
+    Send new request to the Cachito API
+    Send request to check status of existing request
+
+    Checks:
+    * Check that response code is 200
+    * Check in the response that state is complete
+    * Check that “packages” and “dependencies” keys have appropriate values
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response_created_req = client.create_new_request(
+        payload={
+            "repo": test_env["get"]["repo"],
+            "ref": test_env["get"]["ref"],
+            "pkg_managers": test_env["get"]["pkg_managers"],
+        },
+    )
+
+    response = client.fetch_request(response_created_req.id)
+    assert response.status == 200
+
+    while True:
+        if response.data["state"] == ("complete" or "failed"):
+            break
+        time.sleep(5)
+        response = client.fetch_request(response_created_req.id)
+    assert response.data["state"] == "complete"
+
+    assert response.data["packages"][0]["name"] == test_env["get"]["packages_names"]
+
+    response_dependencies_names = set()
+    for data in response.data["dependencies"]:
+        response_dependencies_names = response_dependencies_names.union({data["name"]})
```

Comment:
No need to create a new set for each dependency. Just call `response_dependencies_names.add(data["name"])`.